probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -347,69 +312,23 @@ impl<S: ArmDebugState> SwdSequence for ArmCommunicationInterface<S> {
     }
 }
 
-impl ArmCommunicationInterface<Uninitialized> {
-    pub(crate) fn new(probe: Box<dyn DapProbe>, use_overrun_detect: bool) -> Self {
-        let state = Uninitialized { use_overrun_detect };
-
-        Self {
-            probe: Some(probe),
-            state,
-        }
-    }
-}
-
-impl UninitializedArmProbe for ArmCommunicationInterface<Uninitialized> {
-    fn initialize(
-        mut self: Box<Self>,
-        sequence: Arc<dyn ArmDebugSequence>,
-        dp: DpAddress,
-    ) -> Result<Box<dyn ArmProbeInterface>, (Box<dyn UninitializedArmProbe>, Error)> {
-        let use_overrun_detect = self.state.use_overrun_detect;
-        let probe = self.probe.take().expect("ArmCommunicationInterface is in an inconsistent state. This is a bug, please report it.");
-
-        match ArmCommunicationInterface::<Initialized>::try_setup(
-            probe,
-            sequence,
-            dp,
-            use_overrun_detect,
-        ) {
-            Ok(initialized) => Ok(Box::new(initialized)),
-            Err((probe, err)) => Err((
-                Box::new(ArmCommunicationInterface::new(probe, use_overrun_detect)),
-                Error::Arm(err),
-            )),
-        }
-    }
-
-    fn close(self: Box<Self>) -> Probe {
-        ArmCommunicationInterface::close(*self)
-    }
-}
-
-impl<'interface> ArmCommunicationInterface<Initialized> {
-    /// Set up and start the debug port with brand-new state.
-    fn try_setup(
-        mut probe: Box<dyn DapProbe>,
+impl<'interface> ArmCommunicationInterface {
```

Comment:
What's this lifetime doing for us? I think this is the first time I'm seeing a pattern like this.

Reply:
Not sure why it's defined here, it's only used in one function.

I've moved it there.